src/api/scrapper/extractors.py

```diff
@@ -0,0 +1,68 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup
+from abc import ABC, abstractmethod
+from typing import List
+from api.data.news import Article, TvnArticle, TestArticle
+
+
+class Extractor(ABC):
+    @property
+    @abstractmethod
+    def url(self) -> str:
+        pass
+
+    @abstractmethod
+    def extract(self) -> List[Article]:
+        pass
+
+    def parse_page(self) -> BeautifulSoup:
+        with urlopen(self.url) as response:
+            return BeautifulSoup(response.read(), "html.parser")
+
+
+class TvnExtractor(Extractor):
+    url = "https://tvn24.pl/"
+
+    def extract(self) -> List[TvnArticle]:
+        soup = self.parse_page()
+        articles = []
+
+        css_selector = ".news-of-the-day>ul>li>div>div"
+        result = soup.select(css_selector)
+        for div in result:
+            title = div.get_text()
+            url = div.a['href']
+            articles.append(TvnArticle(title=title, url=url))
+
+        css_selector = ".virtual-page>div.teaser-wrapper>article>div>div"
+        result = soup.select(css_selector)
+        for div in result:
+            if len(div.select("h2")) > 0:
+                title = div.select("h2")[0].get_text()
+                url = div.a['href']
+                articles.append(TvnArticle(title=title, url=url))
+
+        return articles
+
+
+class FakeExtractor(Extractor):
+    url = "fake"
+
+    def extract(self) -> List[TestArticle]:
```

Comment:
Use just Article, no need for using TestArticle if it's for test purposes only

Reply:
OK